lib/ansible/modules/identity/keycloak/keycloak_client.py

```diff
@@ -580,6 +580,7 @@ def main():
                 if module._diff:
                     result['diff'] = dict(before=sanitize_cr(before_client),
                                           after=sanitize_cr(updated_client))
+                result['changed'] = (before_client != updated_client)
```

Comment:
If the client's lists are not sorted (looking at 'protocolMappers' https://www.keycloak.org/docs-api/3.4/rest-api/index.html#_clientrepresentation) this will incorrectly show True. I'm not sure what they're sorted by, but you could add something along the lines of `new_param_value = sorted(new_param_value, key=lambda k: k['id'])` to your `except TypeError` on [line 536](https://github.com/ansible/ansible/pull/37776/files#diff-73a1f8320d09ef0190fcf41558d5ad90L536)

Does the API return optional parameters as empty or omit them if they weren't provided?

Reply:
Ahah. Okay, looking at https://github.com/ansible/ansible/pull/39515/files#diff-73a1f8320d09ef0190fcf41558d5ad90R727, I'm going to take that to mean that protocolMappers are not sorted.

Reply:
Yeah, protocolMappers returned seem to be unsorted (or sorted by a key I do not know). There are other diff problems in check mode I still need to look at (relating to variable typing and optional fields); the more complex a client definition is the higher the chance it will be bitten by these problems at the moment.

In the more trivial cases this change works. I will take another stab at this when doing a pass to add new Keycloak features (4.0.0 just got released).